### PR TITLE
New RPC SetCloudAppProperties

### PR DESCRIPTION
**Pull request by ShobhitAd**
#2215
This PR is **ready** for review.

### Risk
This PR makes **major** API changes.

### Testing Plan
Create ATF test scripts to test this new RPC

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/set_cloudapp_properties_request.cc [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    return;
+  }
+
+  if ((*message_)[strings::msg_params].empty()) {
+    LOG4CXX_ERROR(logger_, strings::msg_params << " is empty.");
+    SendResponse(false, mobile_apis::Result::INVALID_DATA);
+    return;
+  }
+
+  smart_objects::SmartObject cloudapp_properties(smart_objects::SmartType_Map);
```
Comment (JackLivio) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/set_cloudapp_properties_request.cc]: Maybe make cloud app separate words "cloud_app..."

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/set_cloudapp_properties_request.cc [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    return;
+  }
+
+  if ((*message_)[strings::msg_params].empty()) {
+    LOG4CXX_ERROR(logger_, strings::msg_params << " is empty.");
+    SendResponse(false, mobile_apis::Result::INVALID_DATA);
+    return;
+  }
+
+  smart_objects::SmartObject cloud_app_properties(smart_objects::SmartType_Map);
```
Comment (jacobkeeler) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/set_cloudapp_properties_request.cc]: No need to create this object, the policy handler function is designed to just take the raw message object directly.

File: src/components/interfaces/MOBILE_API.xml [hunk 3]
```diff
@@ -545,6 +545,13 @@
         <element name="VEHICLEDATA_CLOUDAPPVEHICLEID" since="5.1"/>
     </enum>
     
+    <enum name="HybridAppPreference" since="5.1">
+        <description>Enumeration for the user's preference of which app type to use when both are available</description>
+        <element name = "MOBILE" />
```
Comment (jacobkeeler) [inline src/components/interfaces/MOBILE_API.xml]: shouldn't be spaces in `name="value"`

**Comment by ShobhitAd** [comment]
Rebased branch from `feature/cloud_app_polcies` to `feature/cloud_app_transport`

**Pull request merged**